### DISCUSSION TITLE
Loot table editor

### DISCUSSION THREAD

**Discussion by PyroTech03** [AscensionGameDev/Intersect-Engine#2030] [Ideas]
Allow us to define loot tables that can be called by NPC's loot drops and within events.

This would potentially save a lot of time and reduce the chances of missing something when trying to add/change loot of given NPC's and expand on capabilities of say world chests.

Allow loot tables to be added like we currently add items, allowing us to even use multiple loot tables on a single entity. Such as we could have a loot table for currency and for gems and for gear, all with their own chance to proc. If proced, refer to the loot tables to see what drops.

On the loot table editor, allow us to define if it's only one item that can drop on it (essentially deciding a weight value) or if multiple can drop from the loot table (like a gem table able to drop multiple types of gems).

It's generic and flexible that many could begin incorporating it and doesn't break current functionality.

**Comment by da-rulez** [comment]
YES! ADDING MY BRIEFING FOR BOAS HERE:

Loot tables: 

New editor to create loot tables
In addition to drops, you can assign a loot table to mobs 
You can assign a range to an amount of items when adding to NPC 
You can assign a % chance of dropping to an item 
System will go down the list and go through each drop chance separately 
If a loot table hits the dropchance, then it will drop one item out of the loottable -> Loot tables are created in a separate editor 

Loot table editor: 
Give name 
Add items 
Give a weight to an item -> This calculates the drop chance. If you have 3 items in the list, with weights: 7,7,6 -> 7/(7+7+6=20) drop chance for first 2 items, 6/20 drop chance for last item

Use cases: 
Easy to add general drops like coins, gems, materials to mobs
Full control over how many items will drop 
Full control over which items will drop


Before: 
Boss NPC -> 1% chance of dropping MAGIC SWORD and 3% chance of dropping SUPER SHIELD and 50% chance of dropping 50 coins -> Possible for boss to drop all 3 items in rare cases 

After: 
Boss NPC -> 50% chance of dropping 2000-5000 coins and 3% chance to hit UBER LOOT TABLE
Uber loot table: 
MAGIC SWORD - Weight 1 
SUPER SHIELD - Weight 3

When you hit the 3% chance of UBER LOOT TABLE -> 1/4 chance you get MAGIC SWORD, 3/4 chance you get SUPER SHIELD

Allow events to roll a loot table as well
